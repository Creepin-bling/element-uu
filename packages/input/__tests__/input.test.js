import input from '../src/input.vue'
import {mount} from '@vue/test-utils'
import { testMatch } from '../../../jest.config'

describe('lg-input',()=>{
    test('input-text',()=>{
        // 挂载组件
        const wrapper=mount(input)
        expect(wrapper.html()).toContain('input type="text')
    })

    test('input-password',()=>{
        const wrapper=mount(input,{
            propsData:{
                type:'password',
            }
        })
        expect(wrapper.html()).toContain('input type="password')
    })

    test('input-password',()=>{
        const wrapper=mount(input,{
            propsData:{
                type:'password',
                value:'admin'
            }
        })
        expect(wrapper.props('value')).toBe('admin')
    })

    // 快照

    test('input-password',()=>{
        const wrapper=mount(input,{
            propsData:{
                type:'password',
                value:'admin'
            }
        })
        expect(wrapper.vm.$el).toMatchSnapshot
    })
})