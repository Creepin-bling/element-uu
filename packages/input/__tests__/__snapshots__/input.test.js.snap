// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lg-input input-password 1`] = `
<div>
  <input
    type="password"
  />
</div>
`;
